### PR TITLE
cli/demo: avoid login URL if insecure

### LINKED ISSUE #58661: CLI: Demo's link to the DB console doesn't work in with --insecure
If you use demo with --insecure on master `~/go/src/github.com/cockroachdb/cockroach$ ./cockroach demo movr --nodes 9 --insecure --with-load` you cannot access the DB Console:
Screen Shot 2021-01-08 at 1 18 10 PM
Screen Shot 2021-01-08 at 1 18 03 PM

However, if you remove the --insecure, you can access the db console.

### PR DESCRIPTION
**Pull request by knz**
Fixes #58661  

NB: insecure is deprecated anyway  #53404

### CODE REVIEW

File: pkg/cli/demo_cluster.go [hunk 1]
```diff
@@ -885,14 +886,16 @@ func (c *transientCluster) listDemoNodes(w io.Writer, justOne bool) {
...
+		if !demoCtx.insecure {
+			// Print node ID and console URL. Embed the autologin feature inside the URL.
+			pwauth := url.Values{
+				"username": []string{c.adminUser.Normalized()},
+				"password": []string{c.adminPassword},
+			}
+			serverURL.Path = server.DemoLoginPath
+			serverURL.RawQuery = pwauth.Encode()
+		}
 		fmt.Fprintf(w, "  (console) %s\n", serverURL)
```
Comment (otan) [inline pkg/cli/demo_cluster.go:899]: i think this still prints the console link, do you mean avoid the link as in avoid printing username and password, or something else?

Comment (knz) [reply pkg/cli/demo_cluster.go:899]: Yes the URL must be printed in any case, what needs to be skipped is the autologin URL path.

File: pkg/cli/demo_cluster.go [hunk 2]
```diff
@@ -885,14 +886,16 @@ func (c *transientCluster) listDemoNodes(w io.Writer, justOne bool) {
...
-		// Print node ID and console URL. Embed the autologin feature inside the URL.
-		pwauth := url.Values{
-			"username": []string{c.adminUser.Normalized()},
-			"password": []string{c.adminPassword},
-		}
 		serverURL := s.Cfg.AdminURL()
-		serverURL.Path = server.DemoLoginPath
-		serverURL.RawQuery = pwauth.Encode()
+		if !demoCtx.insecure {
+			// Print node ID and console URL. Embed the autologin feature inside the URL.
```
Comment (otan) [inline pkg/cli/demo_cluster.go:891]: maybe worth a comment about why we do not do this for insecure

**Review by otan — APPROVED** [review]

File: pkg/cli/demo_cluster.go (hunk 2 above)
Comment (knz) [reply pkg/cli/demo_cluster.go:891]: Sure, done.

**Comment by knz** [comment]
TFYR

bors r=otan

**Comment by knz** [comment]
bors r=otan

**Pull request merged**